src/main/java/org/broadinstitute/hellbender/tools/HtsgetReader.java

```diff
@@ -0,0 +1,441 @@
+            fullName = REFERENCE_LONG_NAME,
+            shortName = REFERENCE_SHORT_NAME,
+            optional = true
+    )
+    private String referenceName;
+
+    @Argument(
+            doc = "Start position of the range on the reference, 0-based, inclusive",
+            fullName = START_LONG_NAME,
+            shortName = START_SHORT_NAME,
+            optional = true
+    )
+    private Integer start;
+
+    @Argument(
+            doc = "The end position of the range on the reference, 0-based exclusive.",
+            fullName = END_LONG_NAME,
+            shortName = END_SHORT_NAME,
+            optional = true
+    )
+    private Integer end;
+
+    @Argument(
+            doc = "A list of fields to include, default: all",
+            fullName = FIELDS_LONG_NAME,
+            shortName = FIELDS_SHORT_NAME,
+            optional = true
+    )
+    private List<HtsgetRequestField> fields;
+
+    @Argument(
+            doc = "A comma separated list of tags to include, default: all.",
+            fullName = TAGS_LONG_NAME,
+            shortName = TAGS_SHORT_NAME,
+            optional = true
+    )
+    private List<String> tags;
+
+    @Argument(
+            doc = "A comma separated list of tags to exclude, default: none.",
+            fullName = NOTAGS_LONG_NAME,
+            shortName = NOTAGS_SHORT_NAME,
+            optional = true
+    )
+    private List<String> notags;
+    
+    @Advanced
+    @Argument(
+            fullName = NUM_THREADS_LONG_NAME,
+            shortName = NUM_THREADS_LONG_NAME,
+            doc = "How many simultaneous threads to use when reading data from an htsget response;" +
+                    "higher values may improve performance when network latency is an issue.",
+            optional = true,
+            minValue = 1)
+    private int readerThreads = 1;
+
+    @Argument(
+            fullName = CHECK_MD5_LONG_NAME,
+            shortName = CHECK_MD5_LONG_NAME,
+            doc = "Boolean determining whether to calculate the md5 digest of the assembled file " +
+                "and validate it against the provided md5 hash, if it exists.",
+            optional = true)
+    private boolean checkMd5 = false;
+
+    private ExecutorService executorService;
+    
+    private CloseableHttpClient client;
+
+    @Override
+    public void onTraversalStart() {
+        if (this.readerThreads > 1) {
+            final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                    .setNameFormat("htsgetReader-thread-%d")
+                    .setDaemon(true)
+                    .build();
+            this.executorService = Executors.newFixedThreadPool(readerThreads, threadFactory);
+        }
+        this.client = HttpClientBuilder.create()
+            .setConnectionManager(new PoolingHttpClientConnectionManager())
+            // throw IOException on 5xx error codes so request is retried
+            .addInterceptorLast(new HttpResponseInterceptor() {
```

Comment:
Changed to use ServiceUnavailableRetryStrategy which allows retrying based on the contents of the HttpResponse, since we're only interested in retrying on 5xx error codes as per the spec, while the default retry handler will handle retrying on IOExceptions etc.